### ISSUE TITLE
[UX] Side-by-side content translation UI

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#2438]
During the local WordPress meetup last week, there was also a TYPO3 presentation. They have a UI where content is being translated to other languages while having the original text displayed next to it:

![typo3-content_translation-languages_side_by_side](https://cloud.githubusercontent.com/assets/2423362/21540532/cd5e9ffc-ce03-11e6-8c2c-a207d07d3918.png)

I realize that we have not yet started working on multilingual support, but there's nothing wrong with planning ahead 😄 

I think that this side-by-side approach is way better than the Drupal way of prepopulating the fields with the text of the original language:

![d8-content_translation](https://cloud.githubusercontent.com/assets/2423362/21541257/07f2e8a8-ce09-11e6-851d-ec5938f57048.png)

That way, when translating content in Drupal, people either overwrite the original text as they are translating, or they start translating below/above it and delete the original text once they are done. Either way, there are UX flaws: if you overwrite the original text, there is no easy way to do a final review - if you keep the original language text around, there's a lot of scrolling up and down.

I thought that perhaps in Backdrop we can have something similar implemented as a two-column layout shipped with core that it is specifically designed for content translation. It would have the content edit form on one column and the preview of the original language on the other side. Perhaps both sides could have the content in edit mode (sometimes you need to adapt the original language text in order to make it easier to be translated in other languages). Perhaps it would also be useful to have a drop-down menu on both sides that would allow selecting which language is on which side. Not entirely sure how we could make this work (or whether it would be possible at all), but on sites with more than two languages, the language drop-down menus could work in an AJAXy way, loading the content edit form of the selected language "on the fly" without page reloads.

Because this would be a layout, people would be able to customize other things as they see fit. For example, changing the layout template to one with 3 columns so that they can be translating/previewing three languages at the same time.

This is just an idea of how things could work UI-wise. Not sure if any of this is possible with the current underlying "infrastructure".

**Comment by klonos** [comment]
Thanks @joejoseph00 for bringing https://www.drupal.org/project/entity_translation_unified_form to my attention in #52. I was not aware of that contrib module (been a while since I last played around with content translation in Drupal/Backdrop):

> Places all entity-translatable fields inline in a single form.

![image](https://user-images.githubusercontent.com/2423362/51793165-26f62f00-2210-11e9-94ae-a93dbd85c0a9.png)

**Comment by joejoseph00** [comment]
Yes I have several clients that are using this module or another similar type custom module.
I convinced the author of this module to publish it on drupal.org, thankfully he did, now we can all enjoy it.

**Comment by joejoseph00** [comment]
NOTE: that module requires the Entity Translation module.

**Comment by klonos** [comment]
>I convinced the author of this module to publish it on drupal.org, thankfully he did, now we can all enjoy it.

Yay contrib, and yay open-source 😄
